docs/github-repository-management.md

```diff
@@ -0,0 +1,142 @@
+
+Much of this configuration is the same across all managed repositories - we decide once the practice that we want to follow, and implement this on all repostories.
+When we add a new repository, change/evolve a decision, or decide on a new common setting, applying everything can involve many PRs or clicking on settings pages (or writing a custom script to update settings).
+This doesn't scale well and takes up precious time that could be spent on more meaningful software development.
+
+## Goal
+
+Provide a single source of truth for configuration of both types (files and settings) that is automatically applied to a list of managed repositories.
+
+### Solution constraints
+
+* Apply updates to managed repositories as Pull Requests
+  * This will allow for CI testing to be run on any changes, ensuring they do not break anything
+  * Automatically approve and merge based on passing CI, so that human intervention is not needed
+* File-based configuration
+  * Keep common configuration files in a single repository (Template Repository)
+  * Allow specifying files to delete from the managed repositories, in addition to the positive case of files to synchronize from the Template Repository
+  * Allow overrides on a per-repo basis to customize their resulting configuration
+  * Allow template files that are rendered to the managed repository, replacing variables specific to the repository
+* Settings configuration
+  * Managed as code, to allow easy inspection and version control
+  * Specified in the same place as file configuration
+  * Any unspecified settings are not touched, so that repositories can have customization where necessary
+  * Allow specifying overrides/extensions of the base set for repositories
+
+Nice to have but not mandatory
+* All infrastructure as GitHub Actions, so that we do not need to enable GitHub Apps or any external infrastructure
+* Settings configuration can be applied from the Template Repository without copying it over to the managed repository
+  * This would avoid having to have a two-step process of "PR to update the settings spec, then have a local action apply it"
+
+## Implementation Notes
+
+There are preexisting projects for some of this functionality (see Appendix)
+During the course of implementation evaluate them, with the following order of preference:
+* if one gets close to meeting the need, consider contributing any remaining features back to the upstream version.
+* If we need to fork a project as a jumping off point, that is second choice
+* Lastly, if nothing comes close enough to meeting our constraints, then we may need to start from scratch on a new action - this will be of clear use to other communities within the ROS ecosystem, and is likely generically useful to other software projects
+
+Application of settings will be performed by having the Template Repository configured as follows
+* Action workflow, triggered on push to the default branch - that runs the file and settings synchronizers
```

Comment:
This part isn't clear to me - does the synchronization need to happen on every push to any ros-tooling repo or just the template repo?